app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ParkingView.java

```diff
@@ -29,4 +34,21 @@ public void showDialogFragment(ListenerDialogFragment listenerDialogFragment) {
     public void showNumberOfParking(String numberOfParking) {
         binding.textViewMainActivity.setText(numberOfParking);
     }
+
+    @Override
+    public void showReservationActivity() {
+        Context context = getContext();
+        if (context != null) {
+            Intent i = new Intent(context, ReservationActivity.class);
+            getActivity().startActivity(i);
```

Comment:
you should never use a getActivity() if you do not validate it is not null. On the other hand, the getActivity() is not necessary, you should use the same context variable